File: frogoui/src/main/java/com/frogobox/ui/fireworks/modifiers/AccelerationModifier.java
package com.frogobox.ui.fireworks.modifiers;

import com.frogobox.ui.fireworks.Particle;

public class AccelerationModifier implements ParticleModifier {

    private final float mVelocityX;
    private final float mVelocityY;

    public AccelerationModifier(float velocity, float angle) {
        float velocityAngleInRads = (float) (angle * Math.PI / 180f);
        mVelocityX = (float) (velocity * Math.cos(velocityAngleInRads));
        mVelocityY = (float) (velocity * Math.sin(velocityAngleInRads));
    }

    @Override
    public void apply(Particle particle, long miliseconds) {
        particle.mCurrentX += mVelocityX * miliseconds * miliseconds;
        particle.mCurrentY += mVelocityY * miliseconds * miliseconds;
    }

}

File: frogoui/src/main/java/com/frogobox/ui/fireworks/modifiers/AccelerationModifier.kt
package com.frogobox.ui.fireworks.modifiers

import com.frogobox.ui.fireworks.Particle
import kotlin.math.cos
import kotlin.math.sin

class AccelerationModifier(velocity: Float, angle: Float) : ParticleModifier {
    private val mVelocityX: Float
    private val mVelocityY: Float

    init {
        val velocityAngleInRads = (angle * Math.PI / 180f).toFloat()
        mVelocityX = (velocity * cos(velocityAngleInRads.toDouble())).toFloat()
        mVelocityY = (velocity * sin(velocityAngleInRads.toDouble())).toFloat()
    }

    override fun apply(particle: Particle?, miliseconds: Long) {
        particle!!.mCurrentX += mVelocityX * miliseconds * miliseconds
        particle!!.mCurrentY += mVelocityY * miliseconds * miliseconds
    }
}